src/engraving/rendering/paddingtable.cpp

```diff
@@ -278,3 +262,151 @@ void PaddingTable::createTable(const MStyle& style)
         elem[ElementType::TAPPING_HALF_SLUR_SEGMENT] = elem[ElementType::ARTICULATION];
     }
 }
+
+double ParenPaddingTable::padding(ElementType type1, ElementType type2)
+{
+    assert(type1 == ElementType::PARENTHESIS || type2 == ElementType::PARENTHESIS);
+    const PaddingVector<double>& row = type1 == ElementType::PARENTHESIS ? m_parenBefore : m_parenAfter;
+    const ElementType& otherType = type1 == ElementType::PARENTHESIS ? type2 : type1;
+
+    return row.at(otherType);
+}
+
+ParenPaddingTablePtr ParenPaddingTable::getPaddingTable(const EngravingItem* parent)
+{
+    ParenPaddingTablePtr table;
+    switch (parent->type()) {
+    case ElementType::NOTE:
+        table = std::make_unique<NoteParenPaddingTable>();
+        break;
+    case ElementType::KEYSIG:
+        table = std::make_unique<KeySigParenPaddingTable>();
+        break;
+    case ElementType::TIMESIG:
+        table = std::make_unique<TimeSigParenPaddingTable>();
+        break;
+    case ElementType::CLEF:
+        table = std::make_unique<ClefParenPaddingTable>();
+        break;
+    default:
+        ASSERT_X("Not a valid parenthesised type")
+    }
+
+    table->setMinimumPaddingUnit(0.1 * parent->style().spatium());
+    table->createTable(parent->style());
```

Comment:
I think this implementation of ParenPaddingTable is way too complex, but I'll refer to our internal discussions with @its-not-nice for that, and I think it's better that we work on this in the next PRs, cause it requires some more trials and testing too.

It's also not optimal to recreate a whole heap-allocated object every single time we query it so it'd be nice to find a way to stash these values. Perhaps not in the same way that we stash the general PaddingTable (cause in many scores the ParenPaddingTable won't even be needed), but maybe creating it the first time it's queried and then stash it for the subsequent calls. But again, if we're going to change the implementation itself better do it in the next PRs.